machine.go

```diff
@@ -590,6 +590,22 @@ func captureFifoToFile(logger *log.Entry, fifoPath string, fifo io.Writer) error
 
 	logger.Debugf("Capturing %q to writer", fifoPath)
 
+	// this goroutine is to track the life of the application along with whether
+	// or not the context has been cancelled. In the event that the context has
+	// been cancel or the application has closed, we will close the fifo file
+	go func() {
+		select {
```

Comment:
This can be written more simply:

```go
select {
	case <-ctx.Done():
	case <-m.exitCh:
}
if err := fifoPipe.Close(); err != nil {
	logger.WithError(err).Debug("failed to close file")
}
```

Reply:
Fixed.